### PR TITLE
Add route for hints on server

### PR DESCRIPTION
**Pull request by JavedNissar**
A a route for requesting hints on the server

### CODE REVIEW

File: server/routes/problem.ts [hunk 1]
```diff
@@ -209,35 +214,92 @@ problem.post<{ name: string }, Submission[], { id: string }>(
     }
 );
 
-problem.post("/:name", async (req, res) => {
+problem.get("/:name", async (req, res) => {
```
Comment (daniyad) [inline server/routes/problem.ts:217]: dont forget to change the endpoint HTTP method on the clinet as well!

File: server/routes/problem.ts [hunk 2]
```diff
@@ -209,35 +214,92 @@ problem.post<{ name: string }, Submission[], { id: string }>(
...
+            res.json({
+                problem_name: problem_name,
+                status: "Runtime Error",
+                error: "user not found",
+            });
+            return;
+        }
+
+        const run = await openai.beta.threads.createAndRun({
+            assistant_id: "asst_7muYZP6iEa04iPJ7FYBJgPiR",
```
Comment (daniyad) [inline server/routes/problem.ts:265]: we should probably put this in .env file

File: server/routes/problem.ts [hunk 3]
```diff
@@ -209,35 +214,92 @@ problem.post<{ name: string }, Submission[], { id: string }>(
...
+            thread: {
+              messages: [
+                { role: "user", content: `Describe how to make this code work: ${code}` },
+              ],
+            },
+          }); 
+        
+        // Sleep for 2 seconds (2000 milliseconds)
```
Comment (daniyad) [inline server/routes/problem.ts:273]: 5 seconds here?

File: server/routes/problem.ts (hunk 1 above)
Comment (JavedNissar) [reply server/routes/problem.ts:217]: Done!

**Review by daniyad — APPROVED** [review]

**Pull request merged**